### PR TITLE
Export test fixes

### PR DESCRIPTION
**Pull request by A5rocks**
Was looking at export tests for https://github.com/python-trio/trio/issues/2885 and spotted a few changes to make!

Refs https://github.com/python-trio/trio/issues/2885

### CODE REVIEW

**Comment by A5rocks** [comment]
pre-commit.ci autofix

**Review by CoolCat467 — APPROVED** [review]
Looks good to me

File: src/trio/_tests/test_exports.py [hunk 1]
```diff
@@ -157,6 +157,9 @@ def no_underscores(symbols: Iterable[str]) -> set[str]:
         ast = linter.get_ast(module.__file__, modname)
         static_names = no_underscores(ast)  # type: ignore[arg-type]
     elif tool == "jedi":
+        if sys.implementation.name != "cpython":
+            pytest.skip("jedi does not support pypy")
```
Comment (jakkdl) [inline src/trio/_tests/test_exports.py:161]:
```suggestion
        if sys.implementation.name != "cpython":  # pragma: no branch
            pytest.skip("jedi does not support pypy")
```

File: src/trio/socket.py [hunk 2]
```diff
@@ -67,13 +67,19 @@
     ntohs as ntohs,
 )
 
+# TODO: update python docs to reflect this
```
Comment (jakkdl) [inline src/trio/socket.py:70]: mostly just this that's a blocker

**Review by jakkdl — APPROVED** [review]

File: src/trio/_tests/test_exports.py (hunk 1 above)
Comment (A5rocks) [reply src/trio/_tests/test_exports.py:161]: This feels weird though, like I don't think codecov should be complaining about this :(

File: src/trio/socket.py (hunk 2 above)
Comment (A5rocks) [reply src/trio/socket.py:70]: Hopefully done with https://github.com/python/typeshed/pull/11073 and https://foss.heptapod.net/pypy/pypy/-/issues/4040.

Comment (A5rocks) [reply src/trio/socket.py]: Well I made a typeshed PR thinking "oh well the stdlib docs are the spec, so PyPy is deviating from it and should not have the docs update for it" but then now I realize `sys.implementation.name` isn't a guard typeshed allows :(

And it looks like these functions just never exist on PyPy. Lol.

**Review by TeamSpen210 — APPROVED** [review]

**Comment by CoolCat467** [comment]
With ca46414 eliminating jakkdl's blocker, I am going to merge this.

**Pull request merged**